src/main.ts

```diff
@@ -48,7 +48,8 @@ class YouTube {
         options.type = options.type || "video";
         
         const filter = options.type === "all" ? "" : `&sp=${Util.filter(options.type)}`;
-        const url = `https://youtube.com/results?q=${encodeURI(query.trim())}&hl=en${filter}`;
+
+        const url = `https://youtube.com/results?q=${encodeURI(query.trim()).replace(/%20/g, '+')}&hl=en${filter}`;
```

Comment:
You can use ``encodeURIComponent(query.trim()).replace(/%20/g, '+')`` for a better support for & / ...